data_steward/cdr_cleaner/cleaning_rules/cancer_concept_suppression.py

```diff
@@ -0,0 +1,166 @@
+"""
+This rule sandboxes and suppresses reccords whose concept_codes end in 
+'History_WhichConditions', 'Condition_OtherCancer', ‘History_AdditionalDiagnosis’,
+and 'OutsideTravel6MonthsWhere'.
+
+Runs on the controlled tier.
+
+Original Issues: DC-1381
+"""
+
+# Python imports
+import logging
+
+# Project imports
+import constants.bq_utils as bq_consts
+from cdr_cleaner.cleaning_rules.base_cleaning_rule import BaseCleaningRule
+from constants.cdr_cleaner import clean_cdr as cdr_consts
+from common import JINJA_ENV, OBSERVATION
+from utils import pipeline_logging
+
+LOGGER = logging.getLogger(__name__)
+
+SANDBOXING_QUERY = JINJA_ENV.from_string("""
+CREATE OR REPLACE TABLE `{{project_id}}.{{sandbox_id}}.{{sandbox_table}}` AS
+SELECT  o.*
+FROM `{{project_id}}.{{dataset_id}}.observation` o
+JOIN `{{project_id}}.{{dataset_id}}.concept` c
+    ON c.concept_id = o.observation_concept_id
+WHERE REGEXP_CONTAINS(c.concept_code, 
+    r'(History_WhichConditions)|(Condition_OtherCancer)|(History_AdditionalDiagnosis)|(OutsideTravel6MonthWhere)'
+)
+""")
```

Comment:
This query might miss some records because `observation_concept_id` only stores the standard_concept, however, the list generated by the regex expression contains non-standard `concept_ids`, therefore those would be missed. 
1. I think this suppression rule is similar to the other ones where the suppression needs to be applied to any *concept_id field just to be sure we are pulling those records out. 
2. The concepts found by regex contain non-standard ones, we might want to include the corresponding standard concepts as well. 

You could take a look at [AbstractConceptSuppression](https://github.com/all-of-us/curation/blob/03170b8cae4675de53a7491fdb2d1c59e8a69708/data_steward/cdr_cleaner/cleaning_rules/deid/concept_suppression.py#L16) and [AbstractBqLookupTableConceptSuppression](https://github.com/all-of-us/curation/blob/03170b8cae4675de53a7491fdb2d1c59e8a69708/data_steward/cdr_cleaner/cleaning_rules/deid/concept_suppression.py#L141). An implementation of this (similar to this one) can be found at [motor_vehicle_accident_suppression.py](https://github.com/all-of-us/curation/blob/develop/data_steward/cdr_cleaner/cleaning_rules/deid/motor_vehicle_accident_suppression.py). It checks a list of suppressed concepts across all *concept_id fields.